client-runtime/src/main/java/com/microsoft/rest/v2/util/FlowableUtil.java

```diff
@@ -70,6 +72,70 @@ public static boolean isFlowableByteBuffer(TypeToken entityTypeToken) {
 
         });
     }
+    private static final class CountingOperator implements FlowableOperator<ByteBuffer, ByteBuffer> {
+        final long bytesExpected;
+        CountingOperator(long bytesExpected) {
+            if (bytesExpected < 0) {
+                throw new IllegalArgumentException("bytesExpected must be non-negative, but was " + bytesExpected);
+            }
+
+            this.bytesExpected = bytesExpected;
+        }
+
+        @Override
+        public Subscriber<? super ByteBuffer> apply(Subscriber<? super ByteBuffer> observer) {
+            final AtomicLong bytesRead = new AtomicLong();
+            final AtomicReference<Subscription> subscription = new AtomicReference<>();
+            return new Subscriber<ByteBuffer>() {
+                @Override
+                public void onSubscribe(Subscription s) {
+                    subscription.set(s);
+                    observer.onSubscribe(s);
+                }
+
+                @Override
+                public void onNext(ByteBuffer byteBuffer) {
+                    long newBytesRead = bytesRead.addAndGet(byteBuffer.remaining());
+                    if (newBytesRead > bytesExpected) {
+                        subscription.get().cancel();
+                        observer.onError(new IllegalArgumentException("Flowable<ByteBuffer> emitted more bytes than the expected " + bytesExpected));
+                    } else {
+                        observer.onNext(byteBuffer);
+                    }
+                }
+
+                @Override
+                public void onError(Throwable t) {
+                    observer.onError(t);
+                }
+
+                @Override
+                public void onComplete() {
+                    long count = bytesRead.get();
+                    if (count != bytesExpected) {
```

Comment:
Should you be explicit here and do a `<` check instead of `!=`?

Reply:
Needs to reject on both the "too few" case and the "too many" case